rust_icu_uloc/src/lib.rs

```diff
@@ -267,6 +332,91 @@ impl ULoc {
         let asciiz = self.as_c_str();
         buffered_string_char_star(uloc_method, asciiz.as_ptr())
     }
+
+    /// Call a `uloc` method that takes this locale's ID, panics on any errors, and returns
+    /// `Some(result)` if the result is non-empty, or `None` otherwise.
+    fn call_buffered_string_method_to_option(
+        &self,
+        uloc_method: unsafe extern "C" fn(
+            *const raw::c_char,
+            *mut raw::c_char,
+            i32,
+            *mut UErrorCode,
+        ) -> i32,
+    ) -> Option<String> {
+        let value = self.call_buffered_string_method(uloc_method).unwrap();
+        if value.is_empty() {
+            None
+        } else {
+            Some(value)
+        }
+    }
+}
+
+/// This implementation is based on ULocale.compareTo from ICU4J.
```

Comment:
Why are we rolling our own?  Are there no ICU4C equivalents?

Reply:
There's nothing in the C API, and the C++ `icu::Locale` defines several operator overrides, but not `<` or `>`.